WordPress/gutenberg
# Split navigation block view script

## What?
This PR contains changes to the navigation block's `viewScript`. These were initially a part of #36176, but it was decided to split them to a separate PR for clarity and separation of concerns.

## Why?

In the navigation block, we only need the modals script if the user has chosen to use modals. In this PR, we're splitting the modals script to a new file, and conditionally load this separate file if it is required.
This improves client-side performance by only loading the things that are needed, and reduces the amount of transferred bytes.

## How?

Uses the implementation from #36176 to allow loading the scripts when needed.

## Testing Instructions
* Add a navigation block and confirm that both scripts get enqueued on the frontend
* Disable modals in the block and confirm that the modals script is no longer loaded.
